p2p/net/swarm/swarm_dial.go

```diff
@@ -103,6 +109,9 @@ func (ds *dialsync) Lock(dst peer.ID) (bool, chan struct{}) {
 		return false, wait
 	}
 
+	// consume rate limiting token
+	ds.ratelimit <- struct{}{}
```

Comment:
(except would need to change (or be more complicated) if it's only for fd-per-conn dials)